### ISSUE TITLE
Game crashes when changing the language to japanese

### ISSUE THREAD

**Issue by Steinhagen** [flathub/com.katawa_shoujo.KatawaShoujo#9]
This problem looks like it's an encoding issue. After setting this option and getting the crash, the game will no longer start (since it will try to set the language to japanese).

```
# flatpak run com.katawa_shoujo.KatawaShoujo
pygame 1.9.6
Hello from the pygame community. https://www.pygame.org/contribute.html
UnicodeDecodeError('ascii', '\xe3\x81\x8b\xe3\x81\x9f\xe3\x82\x8f\xe5\xb0\x91\xe5\xa5\xb3', 0, 1, 'ordinal not in range(128)')

Full traceback:
  File "/app/share/renpy/renpy/bootstrap.py", line 289, in bootstrap
    renpy.main.main()
  File "/app/share/renpy/renpy/main.py", line 341, in main
    renpy.display.core.Interface()
  File "/app/share/renpy/renpy/display/core.py", line 1405, in __init__
    self.set_mode()
  File "/app/share/renpy/renpy/display/core.py", line 1637, in set_mode
    if draw.set_mode(virtual_size, physical_size, fullscreen):
  File "gldraw.pyx", line 179, in renpy.gl.gldraw.GLDraw.set_mode
  File "/app/share/renpy/renpy/display/core.py", line 1448, in post_init
    self.set_window_caption(force=True)
  File "/app/share/renpy/renpy/display/core.py", line 1497, in set_window_caption
    pygame.display.set_caption(caption.encode("utf-8"))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 0: ordinal not in range(128)

After initialization, but before game start.
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 0: ordinal not in range(128)
```

**Comment by manzyun** [comment]
@Steinhagen I appreciate you noticing this issue.
If this issue had not been raised, I might have tried to deal with it in lonely.

So, This is a common issue in countries that use non-alphabetic languages. Of course, Japan falls under this issue.

In the following, I will leave a journal of my search for a solution to this issue. I hope it will be useful in resolving this issue.

# Journal of Search for Solution

1. Run `flatpak run --command=sh --devel com.katawa_shoujo.KatawaShoujo` .
    * This command will give you a glimpse of the contents of the package in a shell.
2. Run `env PYTHONIOENCODING=utf-8 python /app/share/renpy.py` in package's shell.
    * ✅ Successful launch  on my environment.
    * ✅ No problem changed language setting in `renpy.py`
3. Run `env PYTHONIOENCODING=utf-8 /app/bin/KatawaShoujo` in package's shell.
    * ❌ Activation failed. Exactly the same error and traceback were output.

##  My Machine Environment

* OS: Ubuntu 21.10 x86_64
* Kernel: 5.14.12-xanmod1
* CPU: AMD FX-8320E
* GPU: AMD ATI Radeon R7 370
* Memory: 3390MiB

**Comment by manzyun** [comment]
I'm back.

# IMHO notes

That's traceback log say "Can't decode bytes to ASCII" :
> ```
> # flatpak run com.katawa_shoujo.KatawaShoujo
> pygame 1.9.6
> Hello from the pygame community. https://www.pygame.org/contribute.html
> UnicodeDecodeError('ascii', '\xe3\x81\x8b\xe3\x81\x9f\xe3\x82\x8f\xe5\xb0\x91\xe5\xa5\xb3', 0, 1, 'ordinal not in range(128)')
> 
> Full traceback:
>   File "/app/share/renpy/renpy/bootstrap.py", line 289, in bootstrap
>     renpy.main.main()
>   File "/app/share/renpy/renpy/main.py", line 341, in main
>     renpy.display.core.Interface()
>   File "/app/share/renpy/renpy/display/core.py", line 1405, in __init__
>     self.set_mode()
>   File "/app/share/renpy/renpy/display/core.py", line 1637, in set_mode
>     if draw.set_mode(virtual_size, physical_size, fullscreen):
>   File "gldraw.pyx", line 179, in renpy.gl.gldraw.GLDraw.set_mode
>   File "/app/share/renpy/renpy/display/core.py", line 1448, in post_init
>     self.set_window_caption(force=True)
>   File "/app/share/renpy/renpy/display/core.py", line 1497, in set_window_caption
>     pygame.display.set_caption(caption.encode("utf-8"))
> UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 0: ordinal not in range(128)
> 
> After initialization, but before game start.
> UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 0: ordinal not in range(128)
> ```


## I Want Decode that's bytes

I look so like that's bytes to decode "かたわ少女" of UTF-8 decoding. In my opinions.

### details

I referenced Japanese web page: https://orange-factory.com/sample/utf8/code3/e3.html

Please looking that's page's anchor `#Hiragana` . I referenced look table :

* `\xe3\x81\x8b` : `か`
* `\xe3\x81\x9f` : `た`
* `\xe3\x82\x8f` : `わ`

Have you notice that?


# I have confuses

* Why renpy or pygame try ASCII decoding?
* How to fix this issue?

Maybe, I can try fix this issue. But, If I were to do it alone, it might take me 3 years. I also have my life of works(not "job").

-----

This may be a poor post, but I hope it helps to solve this issue.

**Comment by FakeShemp** [comment]
Solved with #13
